### DISCUSSION TITLE
Tosa layers error

### DISCUSSION THREAD

**Discussion by Theoo1997** [iree-org/iree#9759] [General]
I am trying to compile a tflite NN (lenet) and run on a bare-metal platform. I build iree from the github repo (source build) as you have here: https://iree-org.github.io/iree/building-from-source/getting-started/. Then I wonted to compile lenet mlir model using the basic command :
${BUILD_DIR}/tools/iree-compile \
    --iree-stream-partitioning-favor=min-peak-memory \
    --iree-hal-target-backends=dylib-llvm-aot \
    --iree-llvm-target-triple=x86_64-pc-linux-elf \
    --iree-llvm-debug-symbols=false \
  ${BASEDIR}/lenet.mlir -o ${BASEDIR}/dynamic_shapes_cpu.vmfb \".

Unfortunately, I had some  errors like : 
/simple_mul.mlir:2:3: error: The following illegal operations still remain: 
	tosa.const (count: 10)
	tosa.fully_connected (count: 3)
	tosa.mul (count: 1)
	tosa.tanh (count: 4)
	tosa.conv2d (count: 2)
	tosa.exp (count: 1)
	tosa.reciprocal (count: 1)
	tosa.reduce_sum (count: 1)
	tosa.avg_pool2d (count: 2)
	tosa.reshape (count: 1)

  func.func @main(%arg0: tensor<?x28x28x1xf32> {iree.identifier = "conv2d_input"}) -> (tensor<?x10xf32> {iree.identifier = "Identity"}) {
  ^
compilation failed

I see that tosa layers aren't recognized by the compiler. How can I solve this problem? Do I need to enable something in the when I build iree? 
Thank you in advance!!

**Comment by ScottTodd** [comment]
It looks like you're missing the `--iree-input-type=tosa` flag in your call to `iree-compile`. I thought we had some hints about that in the error message, but looks like we don't? We've also considered detecting the input mode automatically where possible.

**Reply by Theoo1997** [reply]
I know that it would be something obvious but yet I couldn't find it. Thanks a lot.

**Comment by stellaraccident** [comment]
We once had a pass that would look for certain dialects and suggest a flag but I think it was removed. May want to add that again to help with cases like this.
